core/storage/sqlite3_ondisk.rs

```diff
@@ -806,10 +809,15 @@ pub fn begin_write_btree_page(
         })
     };
     let c = Completion::new(CompletionType::Write(WriteCompletion::new(write_complete)));
-    page_source.write_page(page_id, buffer.clone(), c)?;
-    Ok(())
+    let res = page_source.write_page(page_id, buffer.clone(), c);
+    if res.is_err() {
```

Comment:
Same comment as before: instead of `clone_counter` can we just decrement the counter after `let res`